### PR TITLE
Do now show "Our analytics" in breadcrumbs if the user doesn't have access to it

### LINKED ISSUE #23981: "Our analytics" appears on "Save question" collection search header even when user doesn't have access to it
**Describe the bug**

A user without read permissions to "Our analytics" collection still sees "Our analytics" on the collection search header when saving a question for the first time. Perhaps this copy should be removed, since "our analytics" is simply the root collection for the organisation, and these users don't see our analytics anywhere else. So it doesn't need to be named here.

You can see it in the screenshot below of react devtools that our "Our analytics" appears as part of the Breadcrumbs component.

image

The same applies when you try to move a question:
image

**To Reproduce**

1. Admin > Permissions > Collections - set "Our analytics" to No access for All Users
2. Login as user
3. Create a new question, click "Save".
4. Open the collection drop-down

Related to https://github.com/metabase/metabase/issues/20716

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/23981
Reproduces https://github.com/metabase/metabase/issues/23981
Related to https://github.com/metabase/metabase/pull/23887

The problem is that we display `Our analytics` title in breadcrumbs in `ItemPicker`. This PR uses the same approach as https://github.com/metabase/metabase/pull/23887:
- Tries to load the root collection via API
- If it works, the root collection will be in the redux store
- If it doesn't work, it means that the user doesn't have access to it
- Changes the title of the root breadcrumb depending on the presence of the root collection in the store

Please note that the entire token for the root collection cannot be removed from breadcrumbs because it also works as a button to navigate to the root.

How to test:
- Log in as an admin
- You need to have a non-admin user. If you don't have one, go to Admin -> People and create it.
- Go to Admin -> Permissions -> Collections and change access for `Our analytics` for `All users` to `No access`
- Login in as the regular user
- New -> Question -> People -> Visualize
- Click `Save` and open the collection picker
- Make sure `Our analytics` breadcrumb is replaced with `Collections`

Screenshot 2022-07-27 at 20 39 42

With access:
Screenshot 2022-07-27 at 20 05 39

Without access:
Screenshot 2022-07-27 at 20 04 46

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionPicker.jsx [hunk 1]
```diff
@@ -23,8 +24,6 @@ import {
 import { SEARCH_DEBOUNCE_DURATION } from "metabase/lib/constants";
 import SelectList from "metabase/components/SelectList";
 
-const { isRegularCollection } = PLUGIN_COLLECTIONS;
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionPicker.jsx:26]: It's not safe to destructure mutable imports like this. Not related to the PR through.

File: frontend/src/metabase/entities/collections/collections.js [hunk 2]
```diff
@@ -159,6 +159,9 @@ export function getExpandedCollectionsById(
   // "Our Analytics"
   collectionsById[ROOT_COLLECTION.id] = {
     ...ROOT_COLLECTION,
+    name: collectionsById[ROOT_COLLECTION.id]
+      ? ROOT_COLLECTION.name
+      : t`Collections`,
```
Comment (ranquild) [inline frontend/src/metabase/entities/collections/collections.js:164]: I picked `Collections` since it what we use in the sidebar

Comment (nemanjaglumac) [reply frontend/src/metabase/entities/collections/collections.js:164]: Nice choice and it's automatically be translated :)

**Review by nemanjaglumac — APPROVED** [review]
Checked the test, ran it locally and it works as expected.
Thank you for adding a repro! 🙇

**Pull request merged**